doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -431,16 +526,21 @@ Creating Hints
          hint_info ::= %| {? @natural } {? @one_pattern }
          one_pattern ::= @one_term
 
-      The first form adds each :n:`@qualid` as a hint with the head symbol of the type of
-      :n:`@qualid` to the specified hint databases (:n:`@ident`\s). The cost of the hint is the number of
-      subgoals generated by :tacn:`simple apply` :n:`@qualid` or, if specified, :n:`@natural`. The
+      The first form adds each :n:`@qualid` as a hint with the head symbol of the
+      type of :n:`@qualid` to the specified hint databases (:n:`@ident`\s). The
+      cost of the hint is :n:`@natural` (which can be inside :n:`@hint_info`). The
       associated pattern is inferred from the conclusion of the type of
       :n:`@qualid` or, if specified, the given :n:`@one_pattern`.
 
       If the inferred type
-      of :n:`@qualid` does not start with a product, :tacn:`exact` :n:`@qualid` is added
-      to the hint list.  If the type can be reduced to a type starting with a product,
-      :tacn:`simple apply` :n:`@qualid` is also added to the hints list.
+      of :n:`@qualid` does not start with a product, the command adds :tacn:`exact`
+      :n:`@qualid` to the hint list.  If the inferred type starts with a product,
+      the command adds :tacn:`simple apply` :n:`@qualid` to the hints list.  Note that
+      unlike :tacn:`auto` and :tacn:`eauto`,
+      :tacn:`typeclasses eauto` uses a tactic equivalent to :tacn:`autoapply`,
+      which behaves somewhat differently from :tacn:`simple apply`.
+      Nonetheless, that tactic's debug output and the hint database misleadingly show
+      :tacn:`simple apply`.
```

Comment:
It should be in a warning

Reply:
Maybe a one-sentence warning along the lines of "Tactics printed in the debug output are similar to but not exactly the same as the ones executed by proof search" and then keep the paragraph mostly as is? Either way the change here is an improvement.